healthcare/healthcare/doctype/clinical_procedure/clinical_procedure.py

```diff
@@ -45,6 +47,17 @@ def after_insert(self):
 			frappe.db.set_value('Clinical Procedure', self.name, 'sample', sample_collection.name)
 		self.reload()
 
+		if self.procedure_template:
+			cp_template = frappe.get_doc('Clinical Procedure Template', self.procedure_template)
+			templates = [cp_template.pre_op_nursing_checklist_template, cp_template.post_op_nursing_checklist_template]
+			for template in templates:
+				if not template:
+					continue
+				NursingTask.create_nursing_tasks_from_template(template, 'Clinical Procedure', self.name)
```

Comment:
```suggestion
			cp_templates = frappe.db.get_value('Clinical Procedure Template', self.procedure_template, ['pre_op_nursing_checklist_template', 'post_op_nursing_checklist_template'], as_dict=1)
			NursingTask.create_nursing_tasks_from_template(cp_templates.pre_op_nursing_checklist_template, 'Clinical Procedure', self.name)
			NursingTask.create_nursing_tasks_from_template(cp.templates.post_op_nursing_checklist_template, 'Clinical Procedure', self.name)
```

- We can avoid `get_doc` here. `get_doc` is always expensive than `db.get_value`.
- Unnecessary for loop. Just call the method twice.